packages/zwave-js/src/lib/commandclass/DoorLockCC.ts

```diff
@@ -479,6 +535,43 @@ latch status:       ${status.latchStatus}`;
 		// Remember that the interview is complete
 		if (!hadCriticalTimeout) this.interviewComplete = true;
 	}
+
+	public handleLockOperationNotification(isLocked: boolean): void {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Door Lock"].withOptions({
+			priority: MessagePriority.NodeQuery,
+		});
+
+		const valueDB = node.valueDB;
+		valueDB.setValue(
+			getCurrentModeValueId(this.endpointIndex),
+			isLocked ? DoorLockMode.Secured : DoorLockMode.Unsecured,
+		);
+		if (valueDB.getValue(getDoorSupportedValueId(this.endpointIndex))) {
+			valueDB.setValue(
+				getDoorStatusValueId(this.endpointIndex),
+				isLocked ? "closed" : "open",
+			);
+		}
```

Comment:
The isLocked status, which you're setting based on keypad and manual operation of the lock doesn't tell us anything definitive about whether the door is open or closed.

Most door locks don't actually monitor the status of the door, and forcibly changing the open / closed status is confusing but ultimately harmless (as long as the user knows not to trust it).  However, if there is a lock that DOES monitor and report open / closed status for the door, this workaround would break that functionality.

I think the current behavior of the door status remaining perpetually set to "open" is for the best, because it clues the end-user into the fact that the lock isn't actually tracking this, and the value should be ignored.

Reply:
What about rather than an `open`/`closed` to just drop an `unknown` for this?

Reply:
> doesn't tell us anything definitive about whether the door is open or closed.

That's why I didn't want to do this at first

> I think the current behavior of the door status remaining perpetually set to "open" is for the best, because it clues the end-user into the fact that the lock isn't actually tracking this, and the value should be ignored.

The lock might be tracking it, but it does not tell us until we ask for it.

> What about rather than an `open`/`closed` to just drop an `unknown` for this?

Also an option - or just remove the value because our cached value is now most likely stale.

Alternatively, we could update only the one value we know for sure has changed and poll the status of the other values. I am leaning towards not updating the values and let the user figure it out (if he even cares).

Reply:
Thanks for your input btw. Some things (especially the ones I can't just try myself) just require some discussion to be ironed out.

Reply:
> Alternatively, we could update only the one value we know for sure has changed and poll the status of the other values. I am leaning towards not updating the values and let the user figure it out (if he even cares).

I wouldn't add polling until someone who has a lock that tracks latch and/or door open status requests it.  For everyone else it's just an extra poll that doesn't return useful data.

Reply:
My lock does track if the door is open or closed and this functionality would be really nice. I've been following this issue and would be happy to help with testing, but I would require specific instructions. I don't know how to build and test these changes.

I'm running home assistant in a virtualbox VM using the Z-wave JS to MQTT add-on.

Reply:
I run zwave2mqtt on a separate rpi with docker compose and followed these direction:
https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/docker_custom_builds

Then built it with:
DOCKER_BUILDKIT=1 docker build --build-arg SRC=git-clone-src --build-arg Z2M_BRANCH=master --build-arg ZWJ_BRANCH=map-notification-lock --no-cache -f zwavejs2mqtt/docker/Dockerfile.contrib -t zwavejs2mqtt .

I don't think you can do a custom built with the Z-wave JS to MQTT add-on.